### ISSUE TITLE
a query about binary search local 

### ISSUE THREAD

**Issue by MangoKiller** [cmhcbb/attackbox#2]
Hello,
I am working through your implementation of the OPT attack,  
If my understand is not wrong , the part of binary_search_local may waste some opportunities of query： 

>      
          if model.predict_label(x0+lbd*theta) == y0:
            lbd_lo = lbd
            lbd_hi = lbd*1.01
            nquery += 1
            while model.predict_label(x0+lbd_hi*theta) == y0:
                lbd_hi = lbd_hi*1.01
                nquery += 1
                if lbd_hi > 20:
                    return float('inf'), nquery
In my opinion, if add _lbd_lo = lbd_hi_ , you know,  just like :

>      
       if model.predict_label(x0+lbd*theta) == y0:
            lbd_lo = lbd
            lbd_hi = lbd*1.01
            nquery += 1
            while model.predict_label(x0+lbd_hi*theta) == y0:
                lbd_lo = lbd_hi 
                lbd_hi = lbd_hi*1.01
                nquery += 1
                if lbd_hi > 20:
                    return float('inf'), nquery
Some opportunities of querying may be saved in the process of binary search.
I wonder if u could clarify this issue for me,
Thanks a miiiiiiiiiiiiillion!

**Comment by MangoKiller** [comment]
Also,  I don't find  **Normalize([0.485....])**   in the process of using CIFAR10, if u could clarify this issue for me, I will very appreciate!

**Comment by cmhcbb** [comment]
Hi, 
  Good suggestion. Yes, in terms of finding better lbd_hi, it does find a better initialization. However, I believe the gain would be really marginal. 
   For the normalization, you could do the normalization for the dataset mean and variance like [0.485...] for a slightly better performance. However, it could be just normalize to image scale to [0,1]. It is easier for us to do the adversarial attack since we don't have clip each dimension separably.

**Comment by MangoKiller** [comment]
Oh I see, thanks for your help~^_^

**Comment by MangoKiller** [comment]
Excuse me, another small question is:
Why you did a lot of record on number of query in [OPT_attack, etc] just like:

>        
    for i in range(num_directions):
             query_count += 1    #<----------just like this
             theta = np.random.randn(*x0.shape)

I mean, you could use PytorchModel.get_num_queries( )
Thanks~

**Comment by cmhcbb** [comment]
Yes, you could. It is the legacy code for recording the query_count. A better way would be just call get_num_queries().

**Comment by MangoKiller** [comment]
I see，again thanks for clarifying
